tests/ha/multi_pod_failure/test_multi_data_pod_failure.py

```diff
@@ -645,103 +618,176 @@ def test_server_data_kpods_fail_during_ios(self):
-        server_pods = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
-        server_pod_name_list = self.system_random.sample(server_pods, self.kvalue)
-        data_pods = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
-        data_pod_name_list = self.system_random.sample(data_pods, self.kvalue)
-        LOGGER.info("Get pod names to be deleted")
-        all_pod_list = server_pod_name_list + data_pod_name_list
-        self.system_random.shuffle(all_pod_list)
-        for count, pod_name in enumerate(all_pod_list):
-            count += 1
-            pod_data = list()
-            pod_data.append(
-                self.node_master_list[0].get_pod_hostname(pod_name=pod_name))  # hostname
-            LOGGER.info("Deleting %s pod %s", count, pod_name)
-            event.set()
-            resp = self.node_master_list[0].delete_deployment(pod_name=pod_name)
-            LOGGER.debug("Response: %s", resp)
-            assert_utils.assert_false(resp[0], f"Failed to delete {count} pod {pod_name} by "
-                                               "deleting deployment (unsafe)")
+        LOGGER.info("Step 5: Perform READs and verify DI on the written data in background")
+        output_rd = Queue()
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': test_prefix_read,
+                'nclients': 1, 'nsamples': 5, 'skipwrite': True, 'skipcleanup': True,
+                'output': output_rd, "setup_s3bench": False, 'event_set_clr': event_set_clr}
+        thread_rd = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        thread_rd.daemon = True  # Daemonize thread
+        thread_rd.start()
+        LOGGER.info("Step 5: Successfully started READs and verified DI on the written data in "
+                    "background")
+        LOGGER.info("Waiting for %s seconds to perform some READs",
+                    HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
 
-            pod_data.append(resp[1])  # deployment_backup
-            pod_data.append(resp[2])  # deployment_name
-            self.restore_pod = self.deploy = True
-            self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+        LOGGER.info("Step 6: Shutdown random %s (K) data & server pods by deleting deployment and "
+                    "verify cluster & remaining pods status", self.kvalue)
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            kvalue=self.kvalue, event=event, event_set_clr=event_set_clr,
+            down_method=const.RESTORE_DEPLOYMENT_K8S, pod_prefix=[const.POD_NAME_PREFIX,
+                                                                  const.SERVER_POD_NAME_PREFIX])
+        # Assert if empty dictionary
```

Comment:
In this test also one-by-one deletion is expected.

Reply:
> In this test also one-by-one deletion is expected.

same as above. IOs are already running and we are deleting pods one by one.

Reply:
same reply as above